### PR TITLE
Fix issue with `onDidChange` not always being called

### LINKED ISSUE #108: onDidChange is called once
Replace fs.watch to fs.watchFile in _watch method.

### PR DESCRIPTION
**Pull request by nhevia**
Fixes #108 and fixes https://github.com/sindresorhus/electron-store/issues/93

Adds a configuration option called `watchFile`.

Works exactly like `watch` but uses `fs.watchFile` instead of `fs.watch`, allowing the subscription to read changes in the file more than once.

```jsx
useEffect(()=>{
    store.onDidChange('myKey', (newV, oldV) => {
      console.log(`Changed from ${oldV} to ${newV}`)
    })
},[])
```

image

Clearly slower (sometimes, benchmarking is unreliable 100/2000ms) than `watch` but the only way to truly watch a file in React (and possible other scenarios, but not tested).

### CODE REVIEW

**Comment by ali-farooq0** [comment]
Great work @nhevia! Really need this to fix https://github.com/sindresorhus/electron-store/issues/93. @sindresorhus can we please get this in ASAP?

**Comment by sindresorhus** [comment]
So `fs.watch` only fails on Ubuntu, right? Can't we just use `fs.watchFile` on Linux? No option needed.

**Comment by sindresorhus** [comment]
Also please open an issue on Node.js. I want to ensure the issue will be eventually fixed in Node.js.

**Comment by ali-farooq0** [comment]
> So `fs.watch` only fails on Ubuntu, right? Can't we just use `fs.watchFile` on Linux? No option needed.

I'm seeing https://github.com/sindresorhus/conf/issues/108 on macOS.

**Comment by sindresorhus** [comment]
> I'm seeing #108 on macOS.

Ok, so let's use `fs.watch` on macOS too then.

**Comment by ali-farooq0** [comment]
you mean `fs.watchFile`? Sure, works for me 👍 . @nhevia are you able to make the necessary changes? It will be a lot nicer to have this work without having to specify an extra option :).

**Comment by sindresorhus** [comment]
Yes

**Comment by nhevia** [comment]
Changes added.

Tests were failing, added to review.

**Review by ali-farooq0 — APPROVED** [review]
Wicked 👍

**Comment by ali-farooq0** [comment]
Thanks for the quick turn around on this @nhevia!

File: test/index.ts [hunk 1]
```diff
@@ -848,7 +848,8 @@ test('.delete() - without dot notation', t => {
 	t.deepEqual(configWithoutDotNotation.get('foo.bar.zoo'), {awesome: 'redpanda'});
 });
 
-const test2 = process.platform === 'darwin' ? test.failing : test;
+/* eslint no-negated-condition: 0 */
```
Comment (sindresorhus) [inline test/index.ts:851]: The lint rule is there for a reason. Don't disable it.

File: test/index.ts [hunk 2]
```diff
@@ -848,7 +848,8 @@ test('.delete() - without dot notation', t => {
...
+const test2 = process.platform !== 'win32' ? test.failing : test;
```
Comment (sindresorhus) [inline test/index.ts:852]: Shouldn't the changes in this PR cause the test to pass on all OSes?

**Comment by sindresorhus** [comment]
The PR title is outdated.

**Comment by ali-farooq0** [comment]
Will you have time to address the feedback @nhevia?

**Comment by sindresorhus** [comment]
@nhevia Bump

**Comment by nhevia** [comment]
Tests passing. Feel free to squash this mess 👀

**Pull request merged**